pylsp/config/schema.json

```diff
@@ -281,7 +281,7 @@
     "pylsp.plugins.pydocstyle.convention": {
       "type": ["string", "null"],
       "default": null,
-      "enum": ["pep257", "numpy", null],
+      "enum": ["pep257", "numpy", "None"],
```

Comment:
Note, I changed the `enum` array value here to include `"None"` instead of `null` as `"None"` was listed as an allowed value in CONFIGURATION.md. But perhaps it should be `["pep257", "numpy", "None", null]` instead?

Reply:
I'd say keep the null as otherwise default will not validate

Reply:
If I make it `["pep257", "numpy", "None", null]` then regenerating the CONFIGURATION.md file results in:

> `string` (one of: `pep257`, `numpy`, `None`, `None`)

Is that an error in the script?

Reply:
Yes, we should probably distinguish between null and None better in the conversion script.

Reply:
> keep the null as otherwise default will not validate

Not sure I understand (be aware I'm only just learning to programme). Where does the default value get validated and wouldn't its value simply be taken as the associated value of the `"default"` key which is `null` in this case anyway?

The value associated with the `"enum"` key is a list (to python) but I don't understand how `null` can be a valid list item since it is not a string. I understand it is a valid item within a JSON array.

I don't understand how the following line takes the item `null` from the list and adds `None` to the `allowed_values` list created with the list comprehension.

https://github.com/python-lsp/python-lsp-server/blob/b218215fbfba125de49b979d67e1239c9477bc5e/scripts/jsonschema2md.py#L44

Incidentally, shouldn't this line include quotes around the allowed string values as the user must set a string value?
```
allowed_values = [f"`\"{value}\"`" for value in prop["enum"]]`
```
Thanks for your patience.

Reply:
> I don't understand how the following line takes the item null from the list and adds None to the allowed_values list created with the list comprehension.

It seems Python when dealing with a JSON array interprets `null` as `None` but otherwise would raise an exception.
```
print(prop["enum"])     # ['pep25', 'numpy', 'None', None]
```

Another misunderstanding I have is why `null` is included in this "enum" array but not in that for `pylsp.configurationSources`. Is it because in that case the `"default"` value is not `null`?

Reply:
Any further thoughts on this? If not I think I will revert it back to how it was but add the quotes in the allowed_values in L44.

Reply:
Yes, please revert this one.

I would say that your idea about adding quotes to denote strings is right. Adding strings for clarity sounds ok, but the implementation should use repr instead (via `!r` formatter) as it will handle more edge cases like escaping and properly denote allowed enum values which are *not* strings:

```
allowed_values = [f"`{value!r}`" for value in prop["enum"]] 
```

Reply:
OK done and thanks I learned about `!r` today. Rebased onto latest `develop` branch. Hope I did so correctly. Thanks again for your time.